src/utils/getAreasList/getAreasList.js

```diff
@@ -21,26 +20,33 @@ export type TAreasList = {
 }
 
 export default function getAreasList(props: TProps): TAreasList {
-  const areas = Object.keys(props).reduce(
+  const propKeys = Object.keys(props)
+
+  /*
+    if the propValue is not a string it is disqualified from being a template
+    from the very get go, mini-optimization there + it removes the need to
+    type-cast the value passed to sanitizeTemplateString since flow can now
+    infer that the value can only be a string otherwise it would have been
+    filtered out already
+  */
+  const templatePropKeys = propKeys.filter(
+    (propKey) =>
+      typeof props[propKey] === 'string' && /template/i.test(propKey),
```

Comment:
Please resolve only for the prop names that *being* from the "template" phrase:

```js
/^template/
```

Also, note that it's case-sensitive. Prop names like `temPlate` must *not* resolve, and must not work.

Reply:
Gotcha. So something like match 'template' case-insensitive, toLowerCase the first letter on that and then match case-sensitive?

Reply:
I believe just `/^template/.test(propKey)` should be enough.

Reply:
but this doesn't check the return value from your parsePropName function since the info returned from that is overkill for this. it just checks the propKey that was passed directly and that will have a capital 'T'.

Reply:
```js
propKey => /^template/.test(propKey) && typeof props[propKey] === 'string'
```
It's okay to have the type check for the prop value, I didn't mean to remove that :) Also, it makes more sense to put value type check *after* the prop name check. It's a good practice to structure conditions based on their probability. The one which has a chance to reject sooner should come before the one which has a smaller chance to reject.